core-js/src/federation.ts

```diff
@@ -0,0 +1,61 @@
+import { BuiltIns, Schema, DirectiveDefinition, NonNullType } from "./definitions";
+
+// TODO: Need a way to deal with the fact that the _Entity type is built after validation.
+export class FederationBuiltIns extends BuiltIns {
```

Comment:
I don't think we should consider the Federation directives + types as built-in. I suspect this is where we'll want to introduce some notion of `Feature` to take care of the namespace mapping in core schemas. We still need to support the current composition behavior when dealing with subgraphs that do not use core schema, but perhaps we can handle that through a more general mechanism as well (optionally marking a feature as built-in?).

Reply:
I think we need both `Feature` and built-ins in some fashion (reminder that this PR doesn't pretend to be ready). And for good or bad, we do need to support existing federation directives as "built-in" for backward compatibility (I mean built-in in the very practical sense that they exists "magically" without the user having defined them in any way).

So I think we need this, at least for now. Hopefully, in some future version, we'll remove `FederationBuiltIns` and all federation types and directives will comes through `Feature`s, but we're not there yet.

I _will_ note that I include `@inacessible` in there and we may not want to expose that one as "built-in" and instead only support it as a feature. So please don't read anything into that one other than "this is work in progress".

> but perhaps we can handle that through a more general mechanism as well

Sure, maybe once we add `Feature` support in this, we've want to refactor a bit, that's ok. But I don't think the way it's currently done is too bad either, because currently, when you use federation, those types and directives _are_ essentially built-in.